docs/shadow-dom.md

```diff
@@ -0,0 +1,42 @@
+# Shadow DOM
+
+The shadow DOM allows us to have styles that are not changed by and do not change the surrounding application. A strong
+boundary like that also comes with a few hurdles.
+
+### Don't turn Shadow DOM off
+
+When you search for shadow DOM related issues you'll often find people arguing just turning it off. Most of the time
+there are proper solutions which you can also find in this repository. The isolation that shadow DOM is providing is
+crucial for a design system and should be used.
+
+As with everything though, there are probably reasonable exceptions to this. But as a general rule shadow DOM should not
+be turned off.
+
+### Inheriting Styles
+
+While the shadow DOM generally prevents styles from piercing through, there are specific exceptions mentioned in the
+[spec](https://www.w3.org/TR/css-scoping-1/#inheritance). Most notably:
+- CSS custom properties
+- color
+- font-*
+- and many others
+
+Check out this [list](https://gist.github.com/dcneiner/1137601) with inherited CSS.
+
+### Use `<slot>`
+
+Slotting elements is a very powerful concept which will help you down the line creating reusable components able to
+be used in various situations.
```

Comment:
In reference to Mika's comment, this could be a good point to (for now) add the example of slotted input elements.